collector/generatorreceiver/helper.go

```diff
@@ -1,25 +1,33 @@
 package generatorreceiver
 
 import (
+	"strings"
 	"encoding/json"
+	"gopkg.in/yaml.v3"
 	"github.com/lightstep/lightstep-partner-sdk/collector/generatorreceiver/internal/topology"
 	"io/ioutil"
 	"os"
 )
 
 func parseTopoFile(topoPath string) (*topology.File, error){
-	var topoFile topology.File
-	jsonFile, err := os.Open(topoPath)
+	var topo topology.File
+	topoFile, err := os.Open(topoPath)
 	if err != nil {
 		return nil, err
 	}
-	defer jsonFile.Close()
+	defer topoFile.Close()
 
-	byteValue, _ := ioutil.ReadAll(jsonFile)
+	byteValue, _ := ioutil.ReadAll(topoFile)
+
+	lowerTopoPath := strings.ToLower(topoPath)
+	if strings.HasSuffix(lowerTopoPath, ".json") {
+		err = json.Unmarshal(byteValue, &topo)
+	} else if strings.HasSuffix(lowerTopoPath, ".yaml") {
+		err = yaml.Unmarshal(byteValue, &topo)
+	}
 
```

Comment:
What happens if we pass a file that is neither json nor yaml?

Reply:
Thanks.  This is updated.

Reply:
meaning: we support `.yml` and we error if the suffix is none of `.json`, `.yaml`, `.yml`